### PR TITLE
feat: support new OpenGemini client with config

### PR DESCRIPTION
**Pull request by Chenxulin97**
support new an OpenGemini client with config

### CODE REVIEW

File: opengemini/client.go [hunk 1]
```diff
@@ -1 +1,71 @@
 package opengemini
+
+import (
+	"crypto/tls"
+	"errors"
+)
+
+type Options = func(client *Client) error
```
Comment (shoothzj) [inline opengemini/client.go:8]: what's Options used for?

File: opengemini/client.go [hunk 2]
```diff
@@ -1 +1,71 @@
...
+
+type Config struct {
+	Addresses   []*Address
```
Comment (shoothzj) [inline opengemini/client.go:11]: Addresses or AddressList?

File: opengemini/client.go [hunk 3]
```diff
@@ -1 +1,71 @@
...
+	AuthConfig  *AuthConfig
+	BatchConfig *BatchConfig
+	GzipEnable  bool
```
Comment (shoothzj) [inline opengemini/client.go:14]: `GzipEnabled bool`

File: opengemini/client.go (hunk 1 above)
Comment (Chenxulin97) [reply opengemini/client.go:8]: sorry I forgot to delete it , I'll modify it later thanx ：）

File: opengemini/client_test.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+					Password: "TestPassword",
+				},
+				BatchConfig: &BatchConfig{
+					BatchEnable:   true,
+					BatchSize:     10,
+					BatchInterval: -1,
+				},
+			},
+			client: nil,
+			err:    errors.New("batch interval and batch size must be more than 0 ,if batch operate enable"),
```
Comment (shoothzj) [inline opengemini/client_test.go:55]:
```suggestion
			err:    errors.New("batch interval and batch size must be more than 0, if batch operate enable"),
```

File: opengemini/client.go [hunk 5]
```diff
@@ -1 +1,52 @@
...
+)
+
+var _, _ = NewClient(&Config{})
+
+type Client interface {
+	WriteBatchPoints(points BatchPoints)
+	WritePoint(point Point)
+	WriteLineProtocol(data, database, retentionPolicy, precision string)
+	Query()
+	Client()
```
Comment (shoothzj) [inline opengemini/client.go:14]: why we need a Client() method?

**Review by xuthus5 — CHANGES_REQUESTED** [review]

**Review by shoothzj — APPROVED** [review]

**Pull request merged**